conans/test/functional/recipe_provides/test_build_requires_conflicts.py

```diff
@@ -0,0 +1,81 @@
+import unittest
+
+from parameterized import parameterized
+
+from conans.test.utils.tools import TestClient, GenConanfile
+
+
+class BuildRequiresTestCase(unittest.TestCase):
+
+    @parameterized.expand([(True,), (False,)])
+    def test_build_require_lib(self, use_single_profile):
+        t = TestClient()
+        t.save({'br_lib.py': GenConanfile("br_lib", "v1").with_provides("libjpeg"),
+                'br.py': GenConanfile("br", "v1").with_require_plain("br_lib/v1"),
+                'app.py': GenConanfile("app", "v1").with_build_require_plain("br/v1")
+                                                   .with_provides("libjpeg")})
+        t.run("create br_lib.py")
+        t.run("create br.py")
+        if use_single_profile:
+            t.run("install app.py", assert_error=True)
+            self.assertIn(" - 'libjpeg' provided by 'app.py (app/v1)', 'br_lib/v1'", t.out)
+        else:
+            t.run("install app.py --profile:host=default --profile:build=default")
+
+    @parameterized.expand([(True,), (False,)])
+    def test_build_require_host(self, use_single_profile):
+        t = TestClient()
+        t.save({'br_lib.py': GenConanfile("br_lib", "v1").with_provides("libjpeg"),
+                'br.py': GenConanfile("br", "v1").with_require_plain("br_lib/v1"),
+                'app.py': GenConanfile("app", "v1").with_build_require_plain("br/v1",
+                                                                             force_host_context=True)
+                                                   .with_provides("libjpeg")})
+        t.run("create br_lib.py")
+        t.run("create br.py")
+        if use_single_profile:
+            t.run("install app.py", assert_error=True)
+        else:
+            t.run("install app.py --profile:host=default --profile:build=default", assert_error=True)
+        self.assertIn(" - 'libjpeg' provided by 'app.py (app/v1)', 'br_lib/v1'", t.out)
```

Comment:
I personally don't like that output differs for different packages (here and line 67):
```
 - 'libjpeg' provided by 'app.py (app/v1)', 'br_lib/v1'
```
```
 - 'libjpeg' provided by 'br_lhs/v1', 'br_rhs/v1'
```
sometimes we print .py filename, sometimes we don't. but as @jgsogo said, it has nothing to do with PR, it's implementation of display name function

Reply:
The ``app.py (app/v1)`` is actually very useful, because allows to differentiate the local conanfile.py in user folder, which is different to the cache ones.